From 26a3e1e96a9350712d03f414345c77d5de8fe1f8 Mon Sep 17 00:00:00 2001
From: Theodor Gherzan <theodor@resin.io>
Date: Thu, 21 Jan 2016 14:16:11 +0100
Subject: resin-extend-expand: Fix the expand bound to a hard limit

[THIS IS MEANT AS A TEMPORARY FIX, NEEDS REVIEW]
The sd driver that ts-7700 provides will report any attached sd device as
being 512G. Due to this our partition tries to expand beyond the real
capacity of the SD card.

Signed-off-by: Theodor Gherzan <theodor@resin.io>
---
 resin-extend-expand | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/resin-extend-expand b/resin-extend-expand
index 8bb92a2..ab95af3 100644
--- a/resin-extend-expand
+++ b/resin-extend-expand
@@ -10,11 +10,11 @@ datadev=$(lsblk $datapart -n -o PKNAME)
 for freespace in $(parted -m /dev/$datadev unit MiB print free | awk -F: '$5 == "free;" {sub("MiB", "", $4); print $4}'); do
     if [ $(echo $freespace'>'$FREESPACE_LIMIT | bc -l) == "1" ]; then
         echo -n "Expand extended partition... "
-        parted -s /dev/$datadev -- resizepart 4 -1s
+        parted -s /dev/$datadev -- resizepart 4 8192
         echo "done."

         echo -n "Expand data partition... "
-        parted -s /dev/$datadev -- resizepart 6 -1s
+        parted -s /dev/$datadev -- resizepart 6 8192
         echo "done."

         partprobe
--
2.1.4

